Documentation/ApiOverview/Fluid/Index.rst

```diff
@@ -0,0 +1,214 @@
+
+You can use Fluid in TYPO3 to do one of the following:
+
+* Create a template (theme) in combination with TypoScript :ref:`FLUIDTEMPLATE <t3tsref:t3tsrefcobj-fluidtemplate>`.
+  Check out the :ref:`t3sitepackage:start` which walks you through the
+  creation of a sitepackage extension.
+* :ref:`adding-your-own-content-elements` in addition to the already existing content
+  elements TYPO3 supplies.
+* The the previous point describes the lightweight components which
+  are created using a combination of TypoScript and Fluid. If you need more functionality
+  or flexibility in your content element, you can create a content plugin using
+  a combination of Extbase and Fluid. This is explained in :ref:`t3extbasebook:start`
+* Use Fluid to create emails using the :ref:`TYPO3 Mail API <mail-fluid-email>`.
+
+Example Fluid snippet
+=====================
+
+This is how a simple Fluid snippet could look like::
+
+   <h4>This is your headline</h4>
+   <p>
+    <f:if condition="{myExpression}">
+      <f:then>
+         {somevariable}
+      </f:then>
+      <f:else>
+          {someothervariable}
+      </f:else>
+    </f:if>
+   </p>
+
+The resulting HTML may look like this::
+
+   <h4>This is your headline</h4>
+   <p>This is the content of variable "somevariable"</p>
+
+The above Fluid snippet contains:
+
+ViewHelpers:
+   The XML elements that start with `f:` like `<f:if>` etc. are ViewHelpers. These
+   are PHP components that are supplied by Fluid and can be used in your Fluid templates.
+   TYPO3 adds some more ViewHelpers for TYPO3 specific functionality. And, you can
+   :ref:`write your own <t3extbasebook:developing-a-custom-viewhelper>`.
+
+   ViewHelpers can do simple processing such as remove spaces with the
+   :ref:`t3viewhelper:typo3fluid-fluid-spaceless` ViewHelper or create a link
+   as is done in the TYPO3 Fluid ViewHelper :ref:`t3viewhelper:link`.
+
+Object Accessors:
+   Fluid can access variables that have been defined. Just use braces
+   and the name of the variable: `{somevariable}`. In Fluid, these placeholders
+   are called `Object Accessors`.
+
+Conditions:
+    The conditions are supplied here by the if / then / else ViewHelpers.
+
+Directory structure
+===================
+
+In your extension, the following directory structure should be used for Fluid files:
+
+.. code-block:: none
+
+   ── Resources
+      └── Private
+        ├── Layouts
+        ├── Partials
+        └── Templates
+
+How you structure the files within the respective `Layouts`, `Partials` and `Templates`
+is up to you. If you are using Extbase controller actions in combination with Fluid,
+Extbase clearly defines how files and directories should be named within these directories,
+see :ref:`extbasebook:template-creation-by-example`.
+
+
+:file:`Templates`
+-----------------
+
+The template contains the main Fluid template. When using a layout (this is optional),
+you must define the sections that are referenced by the layout.
+
+:file:`Layouts`
```

Comment:
Should we have the same "Using Layouts is optional" bold text here as we do in Partials section?

It might help to describe the difference between a template with and without a Layout:

* With a Layout anything that's not inside a section is ignored. When a Layout is used, the Layout determines which sections will be rendered from the template (through use of `f:render` in the Layout file).
* Without a Layout anything that's not inside a section is rendered. You can still use sections of course, but you then must use `f:render` in the template file itself, outside of a section, to render a section.